frontend/packages/dev-console/src/components/pipelines/pipeline-form/PiplelineWorkspacesSection.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import { useFormikContext, FormikValues, useField } from 'formik';
+import FormSection from '../../import/section/FormSection';
+import { DropdownField } from '@console/shared';
+import PVCDropdown from './PVCDropdown';
+import MultipleResourceKeySelector from './MultipleResourceKeySelector';
+import { SecretModel, ConfigMapModel } from '@console/internal/models';
+import { PipelineWorkspace } from 'packages/dev-console/src/utils/pipeline-augment';
+
+export enum VolumeTypes {
+  EmptyDirectory = 'Empty Directory',
+  ConfigMap = 'Config Map',
+  Secret = 'Secret',
+  PVC = 'PVC',
+}
+
+const getVolumeTypeFields = (volumeType: VolumeTypes, index: number) => {
+  switch (VolumeTypes[volumeType]) {
```

Comment:
Ah, I see what happened here... I didn't follow the merits of what you were doing and gave you misguidance. You shouldn't use an enum for a mappable option. It'll make it hard to read back and use the enum to validate the result.

If you're doing a simple "set and forget" model (say like the use of the `PipelineResourceType` enum), you'd be able to use an enum if the keys were the proper option.value and the values could be used as a label. However, if you need to read back the enum and then do a code-based check (like this switch), you're going to have to do weird stuff like this.

Recommend the following actions:

1. Take the enum `VolumeTypes` to `src/components/pipelines/const`.
1. Create an option map that uses the keys of the enum to the current values
1. Use that option map to supply your `DropdownField`
1. Update the type in `pipeline-augment` to use your enum as it's type instead of a string (see my other comments)

```suggestion
  switch (volumeType) {
```

Reply:
I feel that it will increase code redundancy. I cant see the issue with current implementaion of how I am using the enum or with the previous usage of enum as well.

Reply:
This is not how you use an enum. You're overloading concepts for convenience. It's standard to type a variable with an Enum and then just to switch off the value. You're doing this odd "I'm not really using the enum, but I kinda am" logic.

Reply:
Not sure what you meant there, but I don't think I am using enums in a wrong way. I am just reverse mapping the values of the enum as shown here - https://www.typescriptlang.org/docs/handbook/enums.html#reverse-mappings

Reply:
I understand that's how enums _can_ work. Leave it the way it is now, let's just get this merged - my PRs are changing the structure of the Start modal due to other issues and I think it's just easier to have yours merged first.